File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/MaxRule.java
package com.wesleyelliott.kubwa.rule;

/**
 * Created by wesley on 2016/07/28.
 */

public class MaxRule extends Rule<Integer> {

    private int maxValue;

    public MaxRule(Integer maxValue) {
        super(Integer.class);
        this.maxValue = maxValue;
    }

    @Override
    public boolean isValid(Integer value) {
        return value != null && value < maxValue;
    }
}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/MaxRule.kt
package com.wesleyelliott.kubwa.rule

/**
 * Created by wesley on 2016/07/28.
 */

class MaxRule(private val maxValue: Int) : Rule<Int>(Int::class.java) {

    override fun isValid(value: Int?): Boolean {
        return value != null && value < maxValue
    }
}
